src/main/java/com/vint/shop/domain/User.java

```diff
@@ -1,68 +1,92 @@
 package com.vint.shop.domain;
 
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
 import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 import java.time.LocalDate;
+import java.util.Collection;
 import java.util.Objects;
+import java.util.Set;
 
 @Entity
 @Table(name = "\"user\"")
-public class User { //изменить название класса
+public class User implements UserDetails {
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id")
-    private int id;
-    @Column(name = "password_hash")
-    private String password_hash;
+    private long id;
+
+    @NotEmpty(message = "Login cannot be empty")
+    @Size(min = 5, max = 50, message = "Login must be between 5 and 50 characters\n")
+    @Column(name = "login")
+    private String login;
 
-    @Column(name = "first_name")
-    private String first_name;
+    @Column(name = "username")
+    private String username;
 
-    @Column(name = "last_name")
-    private String last_name;
+    @NotEmpty(message = "Password cannot be empty")
+    @Column(name = "password")
+    //@Pattern(regexp = "^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\\s).*$",  message = "\n" + "The password must contain at least 6 characters, at least one number, special character, upper and lower case letter")
+    private String password;
 
+    @Transient //не воспринимается бд
+    private String passwordConfirm;
```

Comment:
Why do you need this field?

Reply:
enter password - confirm password - when registering

Reply:
In this case, it's better to use DTO objects
Ideally, your beans shouldn't contain transient fields